pkg/cas/content_addressable_storage_server_test.go

```diff
@@ -0,0 +1,117 @@
+	digest2 := digest.MustNewDigest("ubuntu1804", "0479688f99e8cbc70291ce272876ff8e0db71a0889daf2752884b0996056b4a0", 234)
+	digest3 := digest.MustNewDigest("ubuntu1804", "7821919ee052d21515cf4e36788138a301c18c36931290270aece8d79ea2cca6", 345)
+
+	request := &remoteexecution.BatchReadBlobsRequest{
+		Digests: []*remoteexecution.Digest{
+			{
+				Hash:      "409a7f83ac6b31dc8c77e3ec18038f209bd2f545e0f4177c2e2381aa4e067b49",
+				SizeBytes: 123,
+			},
+			{
+				Hash:      "0479688f99e8cbc70291ce272876ff8e0db71a0889daf2752884b0996056b4a0",
+				SizeBytes: 234,
+			},
+			{
+				Hash:      "7821919ee052d21515cf4e36788138a301c18c36931290270aece8d79ea2cca6",
+				SizeBytes: 345,
+			},
+		},
+		InstanceName: "ubuntu1804",
+	}
+
+	contentAddressableStorage := mock.NewMockBlobAccess(ctrl)
+
+	a := make([]byte, 123)
+	buf := buffer.NewValidatedBufferFromByteSlice(a)
+	contentAddressableStorage.EXPECT().Get(ctx, digest1).Return(buf)
+	b := make([]byte, 234)
+	buf2 := buffer.NewValidatedBufferFromByteSlice(b)
+	contentAddressableStorage.EXPECT().Get(ctx, digest2).Return(buf2)
+	buf3 := buffer.NewBufferFromError(status.Error(codes.NotFound, "The object you requested could not be found"))
+	contentAddressableStorage.EXPECT().Get(ctx, digest3).Return(buf3)
+
+	contentAddressableStorageServer := cas.NewContentAddressableStorageServer(contentAddressableStorage, 1<<16)
+
+	response, err := contentAddressableStorageServer.BatchReadBlobs(ctx, request)
+	require.NoError(t, err)
+	require.Equal(t, &remoteexecution.BatchReadBlobsResponse{
+		Responses: []*remoteexecution.BatchReadBlobsResponse_Response{
+			{
+				Digest: &remoteexecution.Digest{
+					Hash:      "409a7f83ac6b31dc8c77e3ec18038f209bd2f545e0f4177c2e2381aa4e067b49",
+					SizeBytes: 123,
+				},
+				Data:   a,
+				Status: &status_pb.Status{},
+			},
+			{
+				Digest: &remoteexecution.Digest{
+					Hash:      "0479688f99e8cbc70291ce272876ff8e0db71a0889daf2752884b0996056b4a0",
+					SizeBytes: 234,
+				},
+				Data:   b,
+				Status: &status_pb.Status{},
+			},
+			{
+				Digest: &remoteexecution.Digest{
+					Hash:      "7821919ee052d21515cf4e36788138a301c18c36931290270aece8d79ea2cca6",
+					SizeBytes: 345,
+				},
+				Data: nil,
+				Status: &status_pb.Status{
+					Code:    int32(codes.NotFound),
+					Message: "The object you requested could not be found",
+				},
+			},
+		},
+	}, response)
+}
+
+func TestContentAddressableStorageServerBatchReadBlobsFailure(t *testing.T) {
+	ctrl, ctx := gomock.WithContext(context.Background(), t)
+	defer ctrl.Finish()
+
+	digest1 := digest.MustNewDigest("ubuntu1804", "0000000000000000000000000000000000000000000000000000000000000001", 123)
+	digest2 := digest.MustNewDigest("ubuntu1804", "0000000000000000000000000000000000000000000000000000000000000002", 234)
+
+	request := &remoteexecution.BatchReadBlobsRequest{
+		Digests: []*remoteexecution.Digest{
+			digest1.GetPartialDigest(),
```

Comment:
Could you eliminate the use of GetPartialDigest() here as well?

Reply:
Ah I knew there would be something I'd missed, I forgot to change the hashes of the digests in the second test too.

Reply:
Yeah, the hashes could stay like that if we wanted to. It's not unlikely that some 123 byte blob exists that has that checksum. We don't instantiate any buffers as part of this test, so we're fine.